wormchain/x/tokenfactory/keeper/createdenom.go

```diff
@@ -0,0 +1,95 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	"github.com/wormhole-foundation/wormchain/x/tokenfactory/types"
+)
+
+// ConvertToBaseToken converts a fee amount in a whitelisted fee token to the base fee token amount
+func (k Keeper) CreateDenom(ctx sdk.Context, creatorAddr string, subdenom string) (newTokenDenom string, err error) {
+	denom, err := k.validateCreateDenom(ctx, creatorAddr, subdenom)
+	if err != nil {
+		return "", err
+	}
+
+	err = k.chargeForCreateDenom(ctx, creatorAddr, subdenom)
+	if err != nil {
+		return "", err
+	}
+
+	err = k.createDenomAfterValidation(ctx, creatorAddr, denom)
+	return denom, err
+}
+
+// Runs CreateDenom logic after the charge and all denom validation has been handled.
+// Made into a second function for genesis initialization.
+func (k Keeper) createDenomAfterValidation(ctx sdk.Context, creatorAddr string, denom string) (err error) {
+	denomMetaData := banktypes.Metadata{
+		DenomUnits: []*banktypes.DenomUnit{{
+			Denom:    denom,
+			Exponent: 0,
+		}},
+		Base: denom,
+	}
+
+	k.bankKeeper.SetDenomMetaData(ctx, denomMetaData)
+
+	authorityMetadata := types.DenomAuthorityMetadata{
+		Admin: creatorAddr,
+	}
+	err = k.setAuthorityMetadata(ctx, denom, authorityMetadata)
+	if err != nil {
+		return err
+	}
+
+	k.addDenomFromCreator(ctx, creatorAddr, denom)
+	return nil
+}
+
+func (k Keeper) validateCreateDenom(ctx sdk.Context, creatorAddr string, subdenom string) (newTokenDenom string, err error) {
+	// Temporary check until IBC bug is sorted out
```

Comment:
What is this IBC bug?

Reply:
Took some digging, but it was this bug: https://github.com/cosmos/ibc-go/issues/894 which is present in ibc-go v3.0.0, fixed for ibc-go v4 and backported to ibc-go v3.1.0+.
Since we are using a version without the bug, I have removed the check.